lib/open_food_network/order_cycle_management_report.rb

```diff
@@ -7,22 +9,45 @@ def initialize(user, params = {})
     end
 
     def header
-      ["First Name", "Last Name", "Email", "Phone", "Hub", "Shipping Method", "Payment Method", "Amount"]
+      if is_payment_methods?
+        ["First Name", "Last Name", "Email", "Phone", "Hub", "Shipping Method", "Payment Method", "Amount", "Balance"]
+      else
+        ["First Name", "Last Name", "Delivery Address", "Delivery Postcode", "Phone", "Hub", "Shipping Method", "Payment Method", "Amount", "Balance"]
+      end
     end
 
     def table
-      orders.map do |order|
-        ba = order.billing_address
-        da = order.distributor.andand.address
-        [ba.firstname,
-         ba.lastname,
-         order.email,
-         ba.phone,
-         order.distributor.andand.name,
-         order.shipping_method.andand.name,
-         order.payments.first.andand.payment_method.andand.name,
-         order.payments.first.amount
-        ]
+      if is_payment_methods?
+        orders.map do |order|
+          ba = order.billing_address
```

Comment:
This method is getting really long. That makes it harder to scan, and it's a code smell that suggests that it's time to break it up into component parts. Consider extracting two methods, `payment_method_row(order)` and `delivery_row(order)`. Then this method can be shortened to something like:

```ruby
orders.map do |order|
  if is_payment_methods?
    payment_method_row order
  else
    delivery_row order
  end
end
```